internal/auth/tenant/token_server.go

```diff
@@ -67,6 +72,16 @@ func WithPasswordFlow(v Verifier, validity time.Duration) TokenServerOption {
 	}
 }
 
+func WithPermittedSignatureAlgorithms(applyTo tokenSinkOption) TokenServerOption {
+	return func(ts *TokenServer) {
+		applyTo(ts)
+	}
+}
+
+func (s *TokenServer) GetSignedToken() auth.SignedToken {
+	return s.tokens
+}
```

Comment:
I think this can be greatly simplified
```go
func WithPermittedSignatureAlgorithms(algs []jose.SignatureAlgorithm) TokenServerOptions {
  return func(ts *TokenServer) {
    ts.TokenSource.PermittedSignatureAlgorithms = algs
  }
}
```

You can then delete `auth.SignedToken`, `auth.TokenSink`, `SetPermittedSignatureAlgorithms` and so on.

Reply:
If we want to avoid the jose package leaking into our types we can use `string` instead and do the conversion somewhere lower down.